vcd/resource_vcd_security_tag.go

```diff
@@ -0,0 +1,167 @@
+	}
+}
+
+func resourceVcdOpenApiSecurityTagCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf(errorRetrievingOrg, err)
+	}
+
+	securityTagName := d.Get("name").(string)
+
+	securityTag := &types.SecurityTag{
+		Tag:      securityTagName,
+		Entities: convertSchemaSetToSliceOfStrings(d.Get("vm_ids").(*schema.Set)),
+	}
+	_, err = org.UpdateSecurityTag(securityTag)
+	if err != nil {
+		return diag.Errorf("error when setting up security tags - %s", err)
+	}
+
+	d.SetId(securityTagName) // Security tags doesn't have a real ID. That's why we use the name as ID here.
+
+	return resourceVcdOpenApiSecurityTagRead(ctx, d, meta)
+}
+
+func resourceVcdOpenApiSecurityTagRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf(errorRetrievingOrg, err)
+	}
+
+	securityTagName := d.Id()
+
+	taggedEntities, err := org.GetAllSecurityTaggedEntitiesByName(securityTagName)
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			log.Printf("[DEBUG] Unable to find entities with security tag name: %s. Removing from tfstate", securityTagName)
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error retrieving tagged entities - %s", err)
+	}
+
+	readEntities := make([]string, len(taggedEntities))
+	for i, entity := range taggedEntities {
+		readEntities[i] = entity.ID
+	}
+
+	err = d.Set("vm_ids", convertStringsToTypeSet(readEntities))
+	if err != nil {
+		return diag.Errorf("could not set vm_ids field: %s", err)
+	}
+	return nil
+}
+
+func resourceVcdOpenApiSecurityTagDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf(errorRetrievingOrg, err)
+	}
+
+	securityTagName := d.Id()
+
+	securityTag := &types.SecurityTag{
+		Tag:      securityTagName,
+		Entities: []string{},
+	}
+	_, err = org.UpdateSecurityTag(securityTag)
+	if err != nil {
+		return diag.Errorf("error when deleting security tag - %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceVcdOpenApiSecurityTagImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 2 {
+		return nil, fmt.Errorf("resource name must be specified as org.catalog")
+	}
+
+	orgName, securityTag := resourceURI[0], resourceURI[1]
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgByName(orgName)
+	if err != nil {
+		return nil, fmt.Errorf(errorRetrievingOrg, orgName)
+	}
+
+	filter := url.Values{
+		"filter": []string{"tag==" + securityTag},
+	}
+
+	taggedEntities, err := org.GetAllSecurityTaggedEntities(filter)
```

Comment:
Probably would be more convenient to use the new function `GetAllSecurityTaggedEntitiesByName`